app/context/WalletConnect/helpers.js

```diff
@@ -1,93 +1,54 @@
-import Neon, {
-  rpc,
-  tx,
-  sc,
-  api,
-  wallet,
-  u as uNext,
-} from '@cityofzion/neon-js-next'
+import Neon, { rpc, api } from '@cityofzion/neon-js-next'
 // eslint-disable-next-line
-import { JsonRpcRequest, JsonRpcResponse } from '@json-rpc-tools/utils'
-import { randomBytes } from 'crypto'
+import { JsonRpcResponse } from '@json-rpc-tools/utils'
+import { NeonInvoker } from '@cityofzion/neon-invoker'
+import { NeonSigner } from '@cityofzion/neon-signer'
+import { ContractInvocationMulti } from '@cityofzion/neo3-invoker'
+import { wallet } from '@cityofzion/neon-core'
 import { type SessionRequest } from './WalletConnectContext'
 
-type WitnessScope = {
-  None: 0,
-  /**
-   * CalledByEntry means that this condition must hold: EntryScriptHash == CallingScriptHash
-   * No params is needed, as the witness/permission/signature given on first invocation will automatically expire if entering deeper internal invokes
-   * This can be default safe choice for native NEO/GAS (previously used on Neo 2 as "attach" mode)
-   */
-  CalledByEntry: 1,
-  /**
-   * Custom hash for contract-specific
-   */
-  CustomContracts: 16,
-  /**
-   * Custom pubkey for group members, group can be found in contract manifest
-   */
-  CustomGroups: 32,
-  /**
-   * Global allows this witness in all contexts (default Neo2 behavior)
-   * This cannot be combined with other flags
-   */
-  Global: 128,
-}
-
-type Signer = {
-  scopes: WitnessScope,
-  allowedContracts?: string[],
-  allowedGroups?: string[],
-}
-
-type ContractInvocation = {
-  scriptHash: string,
-  operation: string,
-  args: any[],
-  abortOnFail?: boolean,
-}
-
-export type ContractInvocationMulti = {
-  signers: Signer[],
-  invocations: ContractInvocation[],
-  extraSystemFee?: number,
-  systemFeeOverride?: number,
-  extraNetworkFee?: number,
-  networkFeeOverride?: number,
-}
-
-type SignedMessage = {
-  publicKey: string,
-  data: string,
-  salt: string,
-  messageHex: string,
-}
-
-type SignMessagePayload = {
-  message: string,
-  version: number,
-}
-
 class N3Helper {
   rpcAddress: string
 
   networkMagic: number
 
-  constructor(rpcAddress: string, networkMagic: number) {
+  invoker: NeonInvoker
+
+  signer: NeonSigner
+
+  account: wallet.Account
+
+  constructor(
+    rpcAddress: string,
+    networkMagic: number,
+    invoker: NeonInvoker,
+    signer: NeonSigner,
+    account?: wallet.Account,
+  ) {
+    this.invoker = invoker
+    this.signer = signer
     this.rpcAddress = rpcAddress
     this.networkMagic = networkMagic
+    this.account = account
   }
 
   static init = async (
     rpcAddress: string,
     networkMagic?: number,
-  ): Promise<N3Helper> =>
-    new N3Helper(
+    account?: wallet.Account,
+  ): Promise<N3Helper> => {
+    const invoker = await NeonInvoker.init(rpcAddress, account)
+    const signer = new NeonSigner(account)
+
+    return new N3Helper(
       rpcAddress,
       networkMagic || (await N3Helper.getMagicOfRpcAddress(rpcAddress)),
```

Comment:
We are not using networkMagic and getMagicOfRpcAddress is not needed. You can remove everything related to this. (we talked about this on NWM code review)